### PR TITLE
Adding unit tests for org.atmosphere.util.uri.UriComponent

### PR DESCRIPTION
**Pull request by EdgarScott**
I've analysed your codebase and noticed that `org.atmosphere.util.uri.UriComponent` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important in a subsequent PR.

### CODE REVIEW

File: modules/runtime/src/test/java/org/atmosphere/uri/UriComponentTest.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import java.util.Map;
+
+@RunWith(PowerMockRunner.class)
+public class UriComponentTest {
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void constructorOutputVoid() throws InvocationTargetException,
```
Comment (paulkennethkent) [inline modules/runtime/src/test/java/org/atmosphere/uri/UriComponentTest.java]: I would remove the private constructor test

**Review by paulkennethkent — CHANGES_REQUESTED** [review]
Out of interest there are a lot of inaccessible methods requiring reflection but there must be some kind of public interface where you can construct and access these methods, maybe this is a good case study.

File: modules/runtime/src/test/java/org/atmosphere/uri/UriComponentTest.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(PowerMockRunner.class)
```
Comment (paulkennethkent) [inline modules/runtime/src/test/java/org/atmosphere/uri/UriComponentTest.java]: I don't believe there is any mocking here so you can remove this

File: modules/runtime/src/test/java/org/atmosphere/uri/UriComponentTest.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
+package org.atmosphere.util.uri;
+
+
+import com.diffblue.deeptestutils.Reflector;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.powermock.modules.junit4.PowerMockRunner;
```
Comment (paulkennethkent) [inline modules/runtime/src/test/java/org/atmosphere/uri/UriComponentTest.java]: Remove this import

File: modules/runtime/pom.xml [hunk 4]
```diff
@@ -107,6 +107,36 @@
             <version>20180130</version>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito</artifactId>
+            <version>1.6.5</version>
```
Comment (paulkennethkent) [inline modules/runtime/pom.xml]: After modifying the changes below you can remove this

Comment (paulkennethkent) [reply modules/runtime/pom.xml]: And the other powermock and mockito dependency

**Review by paulkennethkent — CHANGES_REQUESTED** [review]

**Comment by EdgarScott** [comment]
> Out of interest there are a lot of inaccessible methods requiring reflection but there must be some kind of public interface where you can construct and access these methods, maybe this is a good case study.

**Review by paulkennethkent — APPROVED** [review]

**Pull request closed without merging**